pyiceberg/io/pyarrow.py

```diff
@@ -1807,7 +1807,11 @@ def struct(
             else:
                 raise ResolveError(f"Field is required, and could not be found in the file: {field}")
 
-        return pa.StructArray.from_arrays(arrays=field_arrays, fields=pa.struct(fields))
+        return pa.StructArray.from_arrays(
+            arrays=field_arrays,
+            fields=pa.struct(fields),
+            mask=struct_array.is_null() if isinstance(struct_array, pa.StructArray) else None,
```

Comment:
Very interesting, i didnt know about this

https://arrow.apache.org/docs/python/generated/pyarrow.StructArray.html#pyarrow.StructArray.from_arrays
https://arrow.apache.org/docs/python/generated/pyarrow.StructArray.html#pyarrow.StructArray.is_null